python/ml4ir/base/model/architectures/auto_dag_network.py

```diff
@@ -1,7 +1,14 @@
 from __future__ import annotations
 
+from logging import FileHandler
+from pathlib import Path
 from typing import List, Dict, Union, Type
 
+try:
+    import pygraphviz as pgv
+except ImportError:
```

Comment:
@arvindsrikantan You could also mock the import?

Reply:
Were you suggesting to mock here or in the tests? This is not done just for testing. In case the optional dependency is not installed, it has to be ignored.

Reply:
You could mock here too. I don't think that is a better/worse solution than what you have though.